internal/subshell/cmd/env.go

```diff
@@ -69,9 +69,14 @@ func (c *CmdEnv) unset(keyName, oldValue string) error {
 }
 
 func cleanPath(keyValue, oldEntry string) string {
+	oldEntries := make(map[string]bool)
+	for _, entry := range strings.Split(oldEntry, string(os.PathListSeparator)) {
+		oldEntries[filepath.Clean(entry)] = true
+	}
```

Comment:
The environment we get for cleaning is from the config, so doesn't this mean that we're not storing it correctly if we have to do splitting here?

Reply:
For a `state deploy` on Windows, it's adding 4 directories to %PATH%. Since `oldEntry` is a string, it's a ';'-separated list of those directories. We are storing config values correctly in a map -- it's just that the values may be lists like this one.